android/src/main/java/com/tune/react/TuneSDKBridge/TuneSDKBridge.java

```diff
@@ -360,15 +269,33 @@ private void setUserId(Tune tune, String userIdType, String id ) {
      }
 
      private TuneEventItem getTuneEventItem(ReadableMap eventMap ) {
-         return new TuneEventItem(eventMap.getString("itemName"))
-                 .withUnitPrice(eventMap.getDouble("unitPrice"))
-                 .withQuantity(eventMap.getInt("quantity"))
-                 .withRevenue(eventMap.getDouble("revenue"))
-                 .withAttribute1(eventMap.getString("attribute1"))
-                 .withAttribute2(eventMap.getString("attribute2"))
-                 .withAttribute3(eventMap.getString("attribute3"))
-                 .withAttribute4(eventMap.getString("attribute4"))
-                 .withAttribute5(eventMap.getString("attribute5"));
+         TuneEventItem tuneEventItem = new TuneEventItem(eventMap.getString("itemName"));
+
+         if (eventMap.hasKey("attribute1")) {
+             tuneEventItem.withAttribute1(eventMap.getString("attribute1"));
```

Comment:
Suggestion for each of the if-blocks below
```suggestion
             tuneEventItem = tuneEventItem.withAttribute1(eventMap.getString("attribute1"));
```